### PR TITLE
Add getters for AlertEvent

### PR DESCRIPTION
**Pull request by a2l007**
Based out of #5505 
This PR adds getter methods for specific fields of AlertEvent which were removed previously. Although these getter methods did not have any usages in any of the community extensions, they were being used in our internal emitter extensions.
The AlertEvent dimensions could be useful for other emitter extensions as well and could be a @PublicApi in the future.

### CODE REVIEW

**Comment by leventov** [comment]
Please add appropriate `@PublicApi` annotations in this PR. Also, the restored methods should be annotated `@SuppressWarnings("unused")` with comments that they are used in some proprietary extensions.

**Comment by a2l007** [comment]
@leventov `@PublicApi` and other druid specific annotations are not accessible from `java-util`. Would it make sense to move the emitter related code over to `druid-api`

**Comment by leventov** [comment]
It's going to happen in https://github.com/druid-io/druid/pull/5490

**Comment by a2l007** [comment]
Will make the changes once that PR is merged.

**Comment by a2l007** [comment]
@leventov Looks like #5490 isn't happening. Any suggestions on how to proceed with this PR?

**Comment by a2l007** [comment]
@leventov I have made the requested changes. Please review.

File: java-util/src/main/java/io/druid/java/util/emitter/service/AlertEvent.java [hunk 1]
```diff
@@ -106,6 +108,26 @@ public String getFeed()
     return "alerts";
   }
 
+  /*
+   * This method is used in certain proprietary emitter extensions
+   */
+  @PublicApi
```
Comment (leventov) [inline java-util/src/main/java/io/druid/java/util/emitter/service/AlertEvent.java]: `@PublicApi` should be on the class itself, not individual methods

**Comment by leventov** [comment]
In #5505 @gianm mentioned other classed to be annotated

**Comment by a2l007** [comment]
@gianm I've made ServiceMetricEvent and RequestLogEvent `@PublicApi` as you had mentioned in #5505. Could you please review?

**Comment by a2l007** [comment]
@leventov Do you have any additional comments?

**Pull request merged**